third_party/nvfuser/csrc/scheduler/registry.cpp

```diff
@@ -34,24 +34,52 @@ std::deque<std::deque<TensorView*>> tvChains(
   return tv_chains;
 }
 
-bool rejectScheduleForTorchGather(
-    TorchGatherOp* torch_gather,
+// bool rejectScheduleForTorchGather(
+template<typename Op>
+bool rejectScheduleFusionInputRequirement(
+    Op* op,
     ScheduleHeuristic schedule_stragety) {
-  if (!torch_gather->input(0)->isFusionInput()) {
+  std::string warning_message = "First input of ";
+  warning_message += op->getOpString();
+  if (!op->input(0)->isFusionInput()) {
     scheduler_debug_utils::canScheduleRejectReason(
         schedule_stragety,
-        "First input of TorchGatherOp must be fusion input.");
+        (warning_message + " must be fusion input.").c_str());
```

Comment:
Does it need to use `.c_str()`?

Reply:
Fancy, it's a variadic function, we don't even need to concat these strings!